Sephora-M/assignment3/LetterGrid.java

```diff
@@ -0,0 +1,117 @@
+package assignment3;
+
+import java.awt.Point;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.Set;
+
+/**Grid of letters. Contains a method that, given a dictionnary represented as a Trie, allows to find 
+ * all the words that can be formed in the grid. 
+ * 
+ * @author Sephora-M
+ */
+public class LetterGrid {
+	char[][] grid;
+	int X;
+	int Y;
+	
+	public LetterGrid(char[][] grid) {
+		if (grid == null || grid[0].length == 0)
+			throw new IllegalArgumentException("Cannot instatiate an empty or a null grid.");
+		this.grid = grid;
+		Y = grid.length;
```

Comment:
optional: I find using X and Y can be a bit confusing sometimes when dealing with tables. It might be easier to read if you used rowCount and columnCount, but X and Y is fine.

Reply:
yea, I agree.. it actually caused me a bit of a headache because. I will keep in mind for next time!